### DISCUSSION TITLE
Support for Multiple Add Columns in single DDL Statement.

### DISCUSSION THREAD

**Discussion by Shoban05** [googleapis/google-cloud-cpp#8517] [Q&A] [answered]
Do we support the following in Spanner. Not able to run it from console showing error after ','

`
ALTER TABLE coupon_bank 
ADD COLUMN col_name_1 INT64,
ADD COLUMN col_name_2 INT64;
`

Error Message:
`Error parsing Spanner DDL statement: ALTER TABLE coupon_bank ADD COLUMN col_name_1 INT64, ADD COLUMN col_name_2 INT64 : Syntax error on line 2, column 28: Expecting 'EOF' but found ',`

**Comment by devbww** [comment]
Hi @Shoban05,

While I agree that the [ALTER TABLE syntax](https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#alter_table) seems to imply that multiple `ADD COLUMN` actions are allowed, I also agree that it doesn't seem to work in practice.

A workaround is to add the columns in two, separate statements:
```
ALTER TABLE coupon_bank ADD COLUMN col_name_1 INT64;
ALTER TABLE coupon_bank ADD COLUMN col_name_2 INT64;
```
If you want to dig deeper than that, we'll have to talk to the backend folks, as that is where all statement parsing occurs.

**Comment by devbww** [comment]
> If you want to dig deeper than that, we'll have to talk to the backend folks, as that is where all statement parsing occurs.

I've asked that question, and will report back here with what I learn.

**Reply by Shoban05** [reply]
Sure @devbww and thanks for your help. As of now, will use workaround approach.

**Comment by devbww** [comment, marked as answer]
I haven't received a reply from the backend folks, I'm afraid.

However, I note that the error message has changed from ...
```
Error parsing Spanner DDL statement: ... : Syntax error on line 2, column 28: Expecting 'EOF' but found ','
```
to ...
```
For the provided action type of the <ALTER TABLE> statement only single action per statement is allowed.
```
which seems to indicate that the restriction is indeed intentional.

I've asked for the documentation to be clarified.

**Reply by Shoban05** [reply]
Yes that seems to quite helpful as we don't have to look up the documentation for ALTER TABLE, as initial the error points to some issue with query.
